src/types.rs

```diff
@@ -214,29 +214,46 @@ impl MagneticField {
 /// Accelerometer output data rate
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub enum AccelOutputDataRate {
-    /// 1 Hz (High-resolution/Normal/Low-power)
+    /// 1 Hz (all modes)
     Hz1,
-    /// 10 Hz (High-resolution/Normal/Low-power)
+    /// 10 Hz all modes)
     Hz10,
-    /// 25 Hz (High-resolution/Normal/Low-power)
+    /// 25 Hz (all modes)
     Hz25,
-    /// 50 Hz (High-resolution/Normal/Low-power)
+    /// 50 Hz (all modes)
     Hz50,
-    /// 100 Hz (High-resolution/Normal/Low-power)
+    /// 100 Hz (all modes)
     Hz100,
-    /// 200 Hz (High-resolution/Normal/Low-power)
+    /// 200 Hz (all modes)
     Hz200,
-    /// 400 Hz (High-resolution/Normal/Low-power)
+    /// 400 Hz (all modes)
     Hz400,
-    /// 1.344 kHz (High-resolution/Normal)
+    /// 1.344 kHz (only normal and high-resolution modes)
     Khz1_344,
-    /// 1.620 kHz (Low-power)
+    /// 1.620 kHz (only low-power mode)
     Khz1_620LowPower,
-    /// 5.376 kHz (Low-power)
+    /// 5.376 kHz (only low-power mode)
     Khz5_376LowPower,
 }
 
 impl AccelOutputDataRate {
+    /// Create an `AccelOutputDataRate` with the given frequency in Hertz.
+    pub const fn from_freq(hz: u16) -> Option<Self> {
```

Comment:
How about implementing `TryFrom` instead? sounds more idiomatic.

Reply:
`TryFrom<u16>` is a bit ambiguous in what exactly the `u16` represents. Also, `TryFrom` is not `const` currently. Arguably, this should probably be called `from_hertz`.

Reply:
I understand. Could you rename the method `from_hertz`, then? It does sound better.
Also, there is a conflict now that I merged #15.